src/Main.cc

```diff
@@ -113,14 +102,21 @@ void OnUpgrade(uWS::HttpResponse<false>* http_response, uWS::HttpRequest* http_r
 
 // Called on connection. Creates session objects and assigns UUID to it
 void OnConnect(uWS::WebSocket<false, true>* ws) {
-    uint32_t session_id = static_cast<PerSocketData*>(ws->getUserData())->session_id;
-    string address = static_cast<PerSocketData*>(ws->getUserData())->address;
+    auto socket_data = static_cast<PerSocketData*>(ws->getUserData());
+    if (!socket_data) {
+        spdlog::error("Error handling WebSocket connection");
```

Comment:
The error message could be a bit more explicit. I suggest adding something like "Error handling WebSocket connection: can't get address, socket data doesn't exist"